### PR TITLE
fix(include-ex-pat): Do work`exclude_pat` and slapd_inc dir refactor

### PR DESCRIPTION
**Pull request by sticky-note**
* Do work `slapd_include_dir` `exlude_pat` that never worked
* Did some refactor concerning `slapd_include_dir`

### CODE REVIEW

**Comment by myii** [comment]
@sticky-note I decided that we need tests and `semantic-release` configured for this formula.  I'm almost done; I'd like to merge that before this PR, so that we can improve the tests as necessary.

**Comment by sticky-note** [comment]
@myii It's okay for me. rebased #30. Do I have to write tests ?

File: openldap/defaults.yaml [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
   client_pkg: openldap
   server_defaults: /etc/default/slapd
+  server_config: /etc/openldap/slapd.conf
   server_pkg: openldap
   service: slapd
-  user: openldap
-  group: openldap
+  user: ldap
+  group: ldap
+  slapd_include_dir: /etc/ldap/include
```
Comment (myii) [inline openldap/defaults.yaml:13]: When reordering this in #30, I found that there were more exact values shared between `RedHat` and `Suse` than the other two, so I used the value from those as the defaults.  So please take this line and use it for `Debian` and then use the `RedHat`/`Suse` one here.

**Comment by myii** [comment]
@sticky-note I'm just about to send you a review.

**Comment by sticky-note** [comment]
@myii rebased on `saltstack-formulas/master`

File: openldap/osfamilymap.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ Debian:
 RedHat:
   client_pkg: openldap-clients
   server_pkg: openldap-servers
+  slapd_include_dir: /etc/openldap/include
```
Comment (myii) [inline openldap/osfamilymap.yaml:35]: Remove from here and put in `defaults.yaml` instead.

File: openldap/osfamilymap.yaml [hunk 3]
```diff
@@ -45,6 +46,7 @@ RedHat:
 Suse:
   client_pkg: openldap2-client
   server_pkg: openldap2
+  slapd_include_dir: /etc/openldap/include
```
Comment (myii) [inline openldap/osfamilymap.yaml:49]: Remove after it's available in `defaults.yaml`.

File: pillar.example [hunk 4]
```diff
@@ -11,6 +11,7 @@ openldap:
   group: 'openldap'
   slapd_services: 'ldap://127.0.0.1:389/ ldaps:/// ldapi:///'
   slapd_options: '-4'  # add other options here if needed
+  slapd_include_dir: /etc/ldap/include
   includes:
     my_include_file: |
       content
```
Comment (myii) [inline pillar.example:17]: Would you mind changing this (and the lines below) to a small, sample include file that actually works?  Then we can get this part added to the tests later on.

**Comment by myii** [comment]
> Do I have to write tests ?

@sticky-note It's up to you, the framework is there now.  Let me know if you want to discuss this further or if you want this merged after the minor changes.

**Comment by sticky-note** [comment]
@myii  Let's, I'll bring further improvements later

**Comment by myii** [comment]
Brilliant, thanks for the adjustments, @sticky-note.

**Pull request merged**